capstone-project/src/main/angular-webapp/src/app/map/map.component.ts

```diff
@@ -231,6 +231,21 @@ export class MapComponent implements OnInit {
     infoWindowComponent.instance.originalBlobKey = markerData.blobKey;
     infoWindowComponent.changeDetectorRef.detectChanges();
 
+    infoWindowComponent.instance.cancelEvent.subscribe(() => {
+      infoWindowComponent.instance.type = MarkerAction.DISPLAY;
+      if (markerData.blobKey) {
+        this.httpClient.get('/blob-service?' + 'blobAction=' + BlobAction.KEY_TO_BLOB + '&blob-key=' + markerData.blobKey, { responseType: 'blob' })
+          .toPromise()
+          .then((blob) => {
+            infoWindowComponent.instance.imageUrl = MapComponent.getUrlFromBlob(blob)
+            infoWindowComponent.changeDetectorRef.detectChanges();
+          });
```

Comment:
1. Add a comment on what's going on here. 
2. This is this logic is repeating, consider extracting to a helper method.